packages/core/http/core-http-router-server-internal/src/request.ts

```diff
@@ -177,9 +177,14 @@ export class CoreKibanaRequest<
     this.headers = isRealReq ? deepFreeze({ ...request.headers }) : request.headers;
     this.isSystemRequest = this.headers['kbn-system-request'] === 'true';
     this.isFakeRequest = !isRealReq;
+    // set to false if elasticInternalOrigin is explicitly set to false
+    // otherwise check for the header or the query param
     this.isInternalApiRequest =
-      X_ELASTIC_INTERNAL_ORIGIN_REQUEST in this.headers ||
-      Boolean(this.url?.searchParams?.has(ELASTIC_INTERNAL_ORIGIN_QUERY_PARAM));
+      this.url?.searchParams?.get(ELASTIC_INTERNAL_ORIGIN_QUERY_PARAM) === 'false'
+        ? false
+        : X_ELASTIC_INTERNAL_ORIGIN_REQUEST in this.headers ||
+          this.url?.searchParams?.has(ELASTIC_INTERNAL_ORIGIN_QUERY_PARAM);
```

Comment:
This approach makes sense to me and it is the same "workaround" we have in place for specifying versions in the versioned route. We just need to make sure that we do not let this system/special query string value be passed down to actual handlers (see the [removal logic](https://github.com/elastic/kibana/blob/2466a172af66452bdd939dddc56506faba7ffb7a/packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts#L192) in versioned route)